qiskit/utils/deprecation.py

```diff
@@ -37,15 +37,27 @@ def deprecate_arguments(
         Callable: The decorated callable.
     """
 
-    del since  # Will be used in a followup to add deprecations to our docs site.
-
     def decorator(func):
+        func_name = func.__qualname__
```

Comment:
FYI we now use `__qualname__`, which for methods will include the class name, e.g. `Foo.my_method`. This is more informative.